pkg/process/config/config.go

```diff
@@ -359,6 +363,15 @@ func NewAgentConfig(loggerName config.LoggerName, yamlPath, netYamlPath string)
 		cfg.LogLevel = "warn"
 	}
 
+	// get the hostname from the datadog agent if a hostname has not been specified in the config
```

Comment:
From our slack discussion, I would propose:

```
// Fargate is handled as an exceptional case (there is no concept of a host, so we use the ARN in-place).
if fargate.IsFargateInstance() {
	if hostname, err := fargate.GetFargateHost(); err == nil {
		cfg.HostName = hostname
	} else {
		log.Errorf("Cannot get Fargate host: %v", err)
	}
}

// Get the hostname via gRPC from the main agent process
if cfg.HostName == "" {
	cfg.HostName, err = getHostnameFromGRPC(util.GetDDAgentClient)
	if err != nil {
		log.Errorf("failed to get hostname from grpc: %v", err)
	}
}

// If the hostname is not set then we fallback to our own hostname mechanism
if cfg.HostName == "" {
	if hostname, err := getHostname(cfg.DDAgentBin); err == nil {
		cfg.HostName = hostname
	} else {
		log.Errorf("Cannot get hostname: %v", err)
	}
}
```

I think the above code could be cleaned up a bit re: branches / error handling but should give you a sense of what I'm thinking re: priority of things to look at

Reply:
Will update the code to create that path. 
Also as discussed all host name path's including Fargate needs to support client override. An old PR which explains the current behavior that we will need to continue support: https://github.com/DataDog/datadog-process-agent/pull/124